src/materialized/src/bin/materialized.rs

```diff
@@ -284,6 +287,27 @@ fn run() -> Result<(), anyhow::Error> {
         None
     };
 
+    // If --disable-telemetry is present, disable telemetry. Otherwise, if
+    // a TELEMETRY_URL environment variable is set, use that as the telemetry
+    // URL. Otherwise (the defaults), enable the production server for release mode
+    // and disable telemetry in debug mode. This should allow for good defaults (on
+    // in release, off in debug), but also easy development during testing of this
+    // feature via the environment variable.
+    let telemetry_url = if popts.opt_present("disable-telemetry") {
+        None
+    } else {
+        match env::var("TELEMETRY_URL") {
+            Ok(url) => Some(url),
+            Err(_) => {
```

Comment:
For consistency with `MZ_WORKERS` could you explicitly separate the `VarError::NotUnicode` case from the `VarError::NotPresent` case? The former is an error that should bubble.

Reply:
Thoughts on this @mjibson?

Reply:
Shouldn't have pushed before addressing this comment when my day ended yesterday. Done now.